lib/ddtrace/contrib/resque/resque_job.rb

```diff
@@ -10,24 +10,53 @@ module ResqueJob
         def around_perform(*args)
           pin = Pin.get_from(::Resque)
           return yield unless pin && pin.tracer
+
           pin.tracer.trace('resque.job', service: pin.service) do |span|
             span.resource = name
             span.span_type = pin.app_type
             yield
             span.service = pin.service
           end
-        ensure
+        end
+
+        def on_failure_datadog_shutdown!(*args)
+          datadog_shutdown!
+        end
+
+        def after_perform_datadog_shutdown!(*args)
+          datadog_shutdown!
+        end
+
+        def datadog_shutdown!
+          pin = Pin.get_from(::Resque)
           pin.tracer.shutdown! if pin && pin.tracer
         end
       end
     end
   end
 end
 
+Resque.before_first_fork do
+  Datadog::Contrib::Resque.sync_writer = nil
+
+  pin = Datadog::Pin.get_from(Resque)
+  next unless pin && pin.tracer && Datadog.configuration[:resque][:use_sync_writer]
+
+  # Create SyncWriter instance before forking
+  Datadog::Contrib::Resque.sync_writer = if Datadog.configuration[:resque][:use_sync_writer]
```

Comment:
Is `Datadog.configuration[:resque][:use_sync_writer]` not redundant given the `unless` line above that checks the same?